src/main/java/org/jabref/gui/edit/ManageKeywordsViewModel.java

```diff
@@ -0,0 +1,140 @@
+package org.jabref.gui.edit;
+
+import java.util.List;
+import java.util.Optional;
+
+import javafx.beans.property.ObjectProperty;
+import javafx.beans.property.SimpleObjectProperty;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+
+import org.jabref.gui.undo.NamedCompound;
+import org.jabref.gui.undo.UndoableFieldChange;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.logic.specialfields.SpecialFieldsUtils;
+import org.jabref.model.FieldChange;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.Keyword;
+import org.jabref.model.entry.KeywordList;
+import org.jabref.preferences.JabRefPreferences;
+
+import org.fxmisc.easybind.EasyBind;
+
+public class ManageKeywordsViewModel {
+
+    private final List<BibEntry> entries;
+    private final KeywordList sortedKeywordsOfAllEntriesBeforeUpdateByUser = new KeywordList();
+    private final JabRefPreferences preferences;
+    private final ObjectProperty<ManageKeywordsDisplayType> displayType = new SimpleObjectProperty<>(ManageKeywordsDisplayType.CONTAINED_IN_ALL_ENTRIES);
+    private final ObservableList<String> keywords;
+
+    public ManageKeywordsViewModel(JabRefPreferences preferences, List<BibEntry> entries) {
+        this.preferences = preferences;
+        this.entries = entries;
+        this.keywords = FXCollections.observableArrayList();
+
+        EasyBind.subscribe(displayType, this::fillKeywordsList);
+    }
+
+    public ManageKeywordsDisplayType getDisplayType() {
+        return displayType.get();
+    }
+
+    public ObjectProperty<ManageKeywordsDisplayType> displayTypeProperty() {
+        return displayType;
+    }
+
+    private void fillKeywordsList(ManageKeywordsDisplayType type) {
+        keywords.clear();
+        sortedKeywordsOfAllEntriesBeforeUpdateByUser.clear();
+
+        if (type == ManageKeywordsDisplayType.CONTAINED_IN_ALL_ENTRIES) {
+            for (BibEntry entry : entries) {
+                KeywordList separatedKeywords = entry.getKeywords(preferences.getKeywordDelimiter());
+                sortedKeywordsOfAllEntriesBeforeUpdateByUser.addAll(separatedKeywords);
+            }
+        } else if (type == ManageKeywordsDisplayType.CONTAINED_IN_ANY_ENTRY) {
+
+            // all keywords from first entry have to be added
+            BibEntry firstEntry = entries.get(0);
+            KeywordList separatedKeywords = firstEntry.getKeywords(preferences.getKeywordDelimiter());
+            sortedKeywordsOfAllEntriesBeforeUpdateByUser.addAll(separatedKeywords);
+
+            // for the remaining entries, intersection has to be used
+            // this approach ensures that one empty keyword list leads to an empty set of common keywords
+            for (int i = 1; i < entries.size(); i++) {
```

Comment:
Is there a reason why here no advanced for loop is used?
The code looks similar to the other if-branch so I suppose it can be replaced with an advanced for loop